Source/KISAddonConfig.cs

```diff
@@ -22,52 +23,154 @@ sealed class KISAddonConfig : MonoBehaviour {
+    public override void StartLoad ()
+    {
+      ConfigAccessor.ReadFieldsInType(owner.GetType(), owner);
+      ConfigAccessor.ReadFieldsInType(typeof(ModuleKISInventory), instance: null);
+
+      // Set inventory module for every eva kerbal
+      DebugEx.Info("Set KIS config...");
+      nodeSettings = GameDatabase.Instance.GetConfigNode("KIS/settings/KISConfig");
+      if (nodeSettings == null) {
+        DebugEx.Error("KIS settings.cfg not found or invalid !");
+      }
     }
+  }
+
+  class KISPodInventoryLoader: LoadingSystem
+  {
+    public bool done;
 
-    // Kerbal parts.
-    UpdateEvaPrefab(MaleKerbalEva, nodeSettings);
-    UpdateEvaPrefab(FemaleKerbalEva, nodeSettings);
-
-    // Set inventory module for every pod with crew capacity.
-    DebugEx.Info("Loading pod inventories...");
-    foreach (AvailablePart avPart in PartLoader.LoadedPartsList) {
-      if (avPart.name == MaleKerbalEva || avPart.name == FemaleKerbalEva
-          || avPart.name == RdKerbalEva
-          || !avPart.partPrefab || avPart.partPrefab.CrewCapacity < 1) {
-        continue;
+    int loadedPartCount;
+    int loadedPartIndex;
+
+    bool HasPodInventories (Part part)
+    {
+      for (int i = part.Modules.Count; i-- > 0; ) {
+        var inv = part.Modules[i] as ModuleKISInventory;
+        if (inv == null) {
+          continue;
+        }
+        if (inv.invType == ModuleKISInventory.InventoryType.Pod) {
+          return true;
+        }
       }
+      return false;
+    }
 
-      DebugEx.Fine("Found part with CrewCapacity: {0}", avPart.name);
-      for (int i = 0; i < avPart.partPrefab.CrewCapacity; i++) {
-        try {
-          var moduleInventory =
-            avPart.partPrefab.AddModule(typeof(ModuleKISInventory).Name) as ModuleKISInventory;
-          KIS_Shared.AwakePartModule(moduleInventory);
-          var baseFields = new BaseFieldList(moduleInventory);
-          baseFields.Load(nodeSettings.GetNode("EvaInventory"));
-          moduleInventory.podSeat = i;
-          moduleInventory.invType = ModuleKISInventory.InventoryType.Pod;
-          DebugEx.Fine("Pod inventory module(s) for seat {0} loaded successfully", i);
-        } catch {
-          DebugEx.Error("Pod inventory module(s) for seat {0} can't be loaded!", i);
+
+    IEnumerator LoadInventories ()
+    {
+      // Kerbal parts.
+      UpdateEvaPrefab(MaleKerbalEva, nodeSettings);
+      UpdateEvaPrefab(FemaleKerbalEva, nodeSettings);
+
+      // Set inventory module for every pod with crew capacity.
+      DebugEx.Info("Loading pod inventories...");
+      for (loadedPartIndex = 0; loadedPartIndex < loadedPartCount; loadedPartIndex++) {
+        AvailablePart avPart = PartLoader.LoadedPartsList[loadedPartIndex];
+        if (!(avPart.name == MaleKerbalEva || avPart.name == FemaleKerbalEva
+              || avPart.name == RdKerbalEva
+              || !avPart.partPrefab || avPart.partPrefab.CrewCapacity < 1)) {
+          if (HasPodInventories (avPart.partPrefab)) {
```

Comment:
Checking for just *any* module may not be enough. Sometimes people misunderstand how to add the KIS inventory module, and add just one module even though one *per seat* is required. The old approach would fix it, but the new one will not. I'd prefer to be on a safe side and have too many modules in such a case: it won't make any troubles other than taking some extra space in the save file. Having too few modules, on the other hand, will trigger a weird error when transferring the staff (and result in a bug report).

Reply:
Hmm, good point.